<?xml version="1.0" ?>
<!-- This file was generated by robot assembler -->
<robot name="quadrupted_robot"
       xmlns:xi="http://www.w3.org/2001/XInclude"
       xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz"
       xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model"
       xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body"
       xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom"
       xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface"
       xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering"
       xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics"
       xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- xacro properties -->
  <xacro:property name="pi" value="3.1415"/>
  <!-- xacro macro -->
  <xacro:macro name="gazebo_link_reference" params="link_name mu1:=0.6 mu2:=0.6" >
    <gazebo reference="${link_name}">
      <mu1>${mu1}</mu1>
      <mu2>${mu2}</mu2>
    </gazebo>
  </xacro:macro>

  <xacro:macro name="gazebo_transmission" params="joint_name" >
    <transmission name="${joint_name}_trans">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="${joint_name}">
        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      </joint>
      <actuator name="${joint_name}_motor">
        <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>
  </xacro:macro>

  <xacro:macro name="lidar_macro" params="xyz">
    <joint name="lidar_joint" type="continuous">
      <parent link="BASE"/>
      <child link="lidar_link"/>
      <origin xyz="${xyz}" rpx="0 0 0"/>
      <axis xyz="1 0 0"/>
      <limit effort="200" velocity="40"/>
    </joint>
    <transmission name="lidar_joint_trans">
      <type>transmission_interface/SimpleTransmission</type>
      <joint name="lidar_joint">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      </joint>
      <actuator name="lidar_joint_motor">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
      </actuator>
    </transmission>
    <link name="lidar_link">
      <visual>
        <origin xyz="0.04 0 0" rpy="0 ${pi/2} 0"/>
        <geometry>
          <cylinder radius="0.02" length="0.06"/>
        </geometry>
        <material name="black">
          <color rgba="0 0 0 1"/>
        </material>
      </visual>
      <xacro:default_inertial mass="0.1"/>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <cylinder radius="0.02" length="0.06" />
        </geometry>
      </collision>
    </link>
    <gazebo reference="lidar_link">
      <material>Gazebo/Black</material>
      <sensor type="ray" name="lidar">
        <pose>0 0 0 0 0 0</pose>
        <visualize>false</visualize>
        <update_rate>10</update_rate>
        <ray>
          <scan>
            <horizontal>
              <samples>400</samples>
              <resolution>1</resolution>
              <min_angle>${-pi/2}</min_angle>
              <max_angle>${pi/2}</max_angle>
            </horizontal>
          </scan>
          <range>
            <min>0.10</min>
            <max>20.0</max>
            <resolution>0.01</resolution>
          </range>
          <noise>
            <type>gaussian</type>
            <mean>0.0</mean>
            <stddev>0.01</stddev>
          </noise>
        </ray>
        <plugin name="gazebo_ros_lidar_controller" filename="libgazebo_ros_laser.so">
          <topicname>lidar_link/scan</topicname>
          <frameName>lidar_link</frameName>
        </plugin>
      </sensor>
    </gazebo>
  </xacro:macro>

  <xacro:macro name="default_inertial" params="mass">
    <inertial>
      <mass value="${mass}"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </xacro:macro>

  <xacro:lidar_macro xyz="0 0 0"/>

  <!-- link block -->
  <!-- body -->
  <xacro:gazebo_link_reference link_name="BASE" />
  <link name="BASE">
  <inertial>
     <mass value="0.699"/>
     <origin rpy="0 0 0" xyz="-0.021645 0.000218 0.011922"/>
     <inertia ixx="0.001331" ixy="-3.302301e-06" ixz="-0.000222" iyy="0.001146" iyz="-2.177634e-05" izz="0.000874"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 0.000000" rpy="0.00000000 -0.00000000 0"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/body-parts-b.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 0.000000" rpy="0.00000000 -0.00000000 0"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/body-parts-b.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- left eye -->
  <visual>
     <origin xyz="0.005750 -0.036000 0.035000" rpy="-0.00000000 -0.00000000 -${pi}"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.005750 -0.036000 0.035000" rpy="-0.00000000 -0.00000000 -${pi}"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- right eye-->
  <visual>
     <origin xyz="0.005750 0.036000 0.035000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.005750 0.036000 0.035000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- left back leg -->
  <visual>
     <origin xyz="-0.035000 -0.027000 -0.036000" rpy="-1.57079633 0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.035000 -0.027000 -0.036000" rpy="-1.57079633 0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.045000 -0.027000 -0.075750" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.045000 -0.027000 -0.075750" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- right back leg -->
  <visual>
     <origin xyz="-0.035000 0.027000 -0.036000" rpy="-1.57079633 0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.035000 0.027000 -0.036000" rpy="-1.57079633 0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.045000 0.027000 -0.075750" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.045000 0.027000 -0.075750" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- right front leg -->
  <visual>
     <origin xyz="-0.035000 -0.027000 0.036000" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.035000 -0.027000 0.036000" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.045000 -0.027000 0.075750" rpy="-1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.045000 -0.027000 0.075750" rpy="-1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <!-- left front leg -->
  <visual>
     <origin xyz="-0.035000 0.027000 0.036000" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.035000 0.027000 0.036000" rpy="1.57079633 0.00000000 1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.045000 0.027000 0.075750" rpy="-1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.045000 0.027000 0.075750" rpy="-1.57079633 -0.00000000 -1.57079633"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h102.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK0" />
  <link name="LINK0">
  <inertial>
     <mass value="0.115"/>
     <origin rpy="0 0 0" xyz="0.0 -0.01693 -0.017696"/>
     <inertia ixx="5.385323e-05" ixy="6.591829e-22" ixz="-8.587575e-22" iyy="2.381893e-05" iyz="-7.395652e-07" izz="4.539431e-05"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.063250 -0.019000" rpy="0.00000000 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.063250 -0.019000" rpy="0.00000000 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK1" />
  <link name="LINK1">
  <inertial>
     <mass value="0.332"/>
     <origin rpy="0 0 0" xyz="2.739442e-17 0.004313 -0.018548"/>
     <inertia ixx="0.000656" ixy="-4.365575e-20" ixz="0.000409" iyy="0.000915" iyz="2.446988e-06" izz="0.000514"/>
  </inertial>
  <visual>
     <origin xyz="-0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK2" />
  <link name="LINK2">
  <inertial>
     <mass value="0.115"/>
     <origin rpy="0 0 0" xyz="9.267949e-19 -0.01693 -0.017696"/>
     <inertia ixx="5.385323e-05" ixy="4.162468e-21" ixz="-7.197383e-22" iyy="2.381893e-05" iyz="-7.395652e-07" izz="4.539431e-05"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="0.000000 -0.063250 -0.019000" rpy="0.00000000 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.063250 -0.019000" rpy="0.00000000 0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>
  <xacro:gazebo_link_reference link_name="LINK3" />
  <link name="LINK3">
  <inertial>
     <mass value="0.332"/>
     <origin rpy="0 0 0" xyz="2.054581e-17 0.004313 -0.018548"/>
     <inertia ixx="0.000656" ixy="-2.910383e-20" ixz="0.000409" iyy="0.000915" iyz="2.446988e-06" izz="0.000514"/>
  </inertial>
  <visual>
     <origin xyz="-0.000000 -0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 -0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK4" />
  <link name="LINK4">
  <inertial>
     <mass value="0.115"/>
     <origin rpy="0 0 0" xyz="3.398248e-18 -0.01693 -0.017696"/>
     <inertia ixx="5.385323e-05" ixy="-1.398424e-20" ixz="-3.264466e-21" iyy="2.381893e-05" iyz="-7.395652e-07" izz="4.539431e-05"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 -0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 -0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.000000 -0.063250 -0.019000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 -0.063250 -0.019000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK5" />
  <link name="LINK5">
  <inertial>
     <mass value="0.332"/>
     <origin rpy="0 0 0" xyz="2.876414e-17 0.004313 -0.018548"/>
     <inertia ixx="0.000656" ixy="-1.091394e-19" ixz="0.000409" iyy="0.000915" iyz="2.446988e-06" izz="0.000514"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
    <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
    <geometry>
    <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
    </geometry>
  </collision>
</link>

  <xacro:gazebo_link_reference link_name="LINK6" />
  <link name="LINK6">
  <inertial>
     <mass value="0.115"/>
     <origin rpy="0 0 0" xyz="-1.699124e-17 -0.01693 -0.017696"/>
     <inertia ixx="5.385323e-05" ixy="-1.048095e-20" ixz="-3.125447e-21" iyy="2.381893e-05" iyz="-7.395652e-07" izz="4.539431e-05"/>
  </inertial>
  <visual>
     <origin xyz="-0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  <visual>
     <origin xyz="-0.000000 -0.063250 -0.019000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="-0.000000 -0.063250 -0.019000" rpy="-3.14159265 -0.00000000 -3.14159265"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/fr12_h101.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>

  <xacro:gazebo_link_reference link_name="LINK7" />
  <link name="LINK7">
  <inertial>
     <mass value="0.332"/>
     <origin rpy="0 0 0" xyz="4.109163e-17 0.004313 -0.018548"/>
     <inertia ixx="0.000656" ixy="5.820766e-20" ixz="0.000409" iyy="0.000915" iyz="2.446988e-06" izz="0.000514"/>
  </inertial>
  <visual>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </visual>
  <collision>
     <origin xyz="0.000000 0.000000 -0.019000" rpy="0.00000000 -0.00000000 0.00000000"/>
     <geometry>
     <mesh filename="package://robot_assembler/meshes/xm430_dynamixel.dae" scale="1.000000 1.000000 1.000000"/>
     </geometry>
  </collision>
  </link>
  <!-- joint block -->
  <xacro:gazebo_transmission joint_name="JOINT0" />
  <joint name="JOINT0" type="revolute">
     <parent link="BASE"/>
     <child link="LINK0"/>
     <origin xyz="-0.045000 -0.046000 -0.075750" rpy="1.57079633 1.04719755 0.00000000"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT1" />
  <joint name="JOINT1" type="revolute">
     <parent link="LINK0"/>
     <child link="LINK1"/>
     <origin xyz="-0.019000 -0.063250 -0.019000" rpy="0.77482336 -1.57079633 -0.77482336"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT2" />
  <joint name="JOINT2" type="revolute">
     <parent link="BASE"/>
     <child link="LINK2"/>
     <origin xyz="-0.045000 0.008000 -0.075750" rpy="1.57079633 1.04719755 0.00000000"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT3" />
  <joint name="JOINT3" type="revolute">
     <parent link="LINK2"/>
     <child link="LINK3"/>
     <origin xyz="-0.019000 -0.063250 -0.019000" rpy="0.77482336 -1.57079633 -0.77482336"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT4" />
  <joint name="JOINT4" type="revolute">
     <parent link="BASE"/>
     <child link="LINK4"/>
     <origin xyz="-0.045000 -0.008000 0.075750" rpy="-1.57079633 -1.04719755 0.00000000"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT5" />
  <joint name="JOINT5" type="revolute">
     <parent link="LINK4"/>
     <child link="LINK5"/>
     <origin xyz="0.019000 -0.063250 -0.019000" rpy="-2.42017915 1.57079633 -2.42017915"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT6" />
  <joint name="JOINT6" type="revolute">
     <parent link="BASE"/>
     <child link="LINK6"/>
     <origin xyz="-0.045000 0.046000 0.075750" rpy="-1.57079633 -1.04719755 0.00000000"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>
  <xacro:gazebo_transmission joint_name="JOINT7" />
  <joint name="JOINT7" type="revolute">
     <parent link="LINK6"/>
     <child link="LINK7"/>
     <origin xyz="0.019000 -0.063250 -0.019000" rpy="-2.42017915 1.57079633 -2.42017915"/>
     <axis xyz="0 0 1"/>
     <limit effort="200" lower="-3.14159" upper="3.14159" velocity="40"/>
  </joint>

  <!-- Gazebo plugin for ROS Control -->
  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>
</robot>
